core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -2221,23 +2224,26 @@ public void testPaginationForListTables(int numberOfItems) {
             eq(RESTCatalogAdapter.Route.LIST_TABLES),
             eq(ImmutableMap.of("pageToken", "", "pageSize", "10", "namespace", namespaceName)),
             any(),
-            eq(ListTablesResponse.class));
+            eq(ListTablesResponse.class),
+            any());
 
     // verify second request with updated pageToken
     Mockito.verify(adapter)
         .handleRequest(
             eq(RESTCatalogAdapter.Route.LIST_TABLES),
             eq(ImmutableMap.of("pageToken", "10", "pageSize", "10", "namespace", namespaceName)),
             any(),
-            eq(ListTablesResponse.class));
+            eq(ListTablesResponse.class),
+            any());
 
     // verify third request with update pageToken
     Mockito.verify(adapter)
         .handleRequest(
             eq(RESTCatalogAdapter.Route.LIST_TABLES),
             eq(ImmutableMap.of("pageToken", "20", "pageSize", "10", "namespace", namespaceName)),
             any(),
-            eq(ListTablesResponse.class));
+            eq(ListTablesResponse.class),
+            any());
```

Comment:
maybe we should add a simple test that verifies that the correct response headers are being returned

Reply:
Currently RESTCatalogAdapter.handleRequest() doesn't populate the responseHeaders param. As a follow-up PR I'm planning to add functionality to RESTCatalogAdapter so that it can take care of the header and return codes required for freshness aware loading (like the 'if-None-Match' input header, 'ETag' output header and the 304 response).
This current PR is just a boilerplate code to have the request and response headers available in the get-post requests for a later use.

Reply:
@nastra do you think the above makes sense?